truffle/src/main/java/org/jruby/truffle/runtime/rope/LeafRope.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2015, 2016 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 1.0
+ * GNU General Public License version 2
+ * GNU Lesser General Public License version 2.1
+ */
+package org.jruby.truffle.runtime.rope;
+
+import org.jcodings.Encoding;
+
+import java.util.Arrays;
+
+public abstract class LeafRope extends Rope {
+
+    private final byte[] bytes;
```

Comment:
I don't think it matters much currently because it's invoked through a polymorphic method and that seems to be an implicit boundary to Graal. I need to lift the field into the base `Rope` class, along with the accessor method.

I did try that and ran into some problems. I think it was because some ropes still lazily populate the byte[]. It's on my list to revisit this as part of working out whether to share the byte[] or eagerly allocate it. In either case, I want it to be a final field for all ropes and then I'll mark it `@CompilationFinal`.